### PR TITLE
8583 nested divisions public site

### PR DESCRIPTION
**Pull request by arisah-ijeoma**

### CODE REVIEW

**Comment by emitche** [comment]
@arisah-ijeoma Do you know why the tests are failing? Are you able to fix them at this time? Thank you!

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -72,4 +72,28 @@ def content_tag_if(condition, *args, &block)
   def app_version_number
     @app_version ||= File.read(File.join(Rails.root, "VERSION"))
   end
+
+  def division_select_options(include_root: true, include_all: false, public: false)
+    divisions = public ? Division.published : current_user.accessible_divisions
```
Comment (meltheadorable) [inline app/helpers/application_helper.rb]: I think these calls should really be wrapped in a `policy_scope` call -- we don't want to be setting up any public-facing records that haven't been filtered through the policy/authorization system

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -72,4 +72,28 @@ def content_tag_if(condition, *args, &block)
   def app_version_number
     @app_version ||= File.read(File.join(Rails.root, "VERSION"))
   end
+
+  def division_select_options(include_root: true, include_all: false, public: false)
```
Comment (meltheadorable) [inline app/helpers/application_helper.rb]: Can we default to the more restrictive option (`public_only: true` [and also change `public` to `public_only`]) and change the other calls to match? 

In general it's a best practice, when you're dealing with collections that can either be more or less restricted, to make the default the more restricted option, that way there's less of a chance that you'll leak any extra information.

Comment (arisah-ijeoma) [reply app/helpers/application_helper.rb]: thank you

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (arisah-ijeoma) [reply app/helpers/application_helper.rb]: 👍

**Pull request merged**